### PR TITLE
disallow shipping to unsuported countries

### PR DESCRIPTION
**Pull request by sjolicoeur**

### CODE REVIEW

File: paypal/express/views.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def _get_redirect_url(self, basket, **kwargs):
 
                 shipping_method = self.get_shipping_method(
                     basket, shipping_addr)
+                c
```
Comment (monokrome) [inline paypal/express/views.py]: @sjolicoeur This will break the interpretter

File: paypal/express/views.py [hunk 2]
```diff
@@ -71,15 +71,15 @@ def get_redirect_url(self, **kwargs):
         except InvalidBasket as e:
             messages.warning(self.request, six.text_type(e))
             logger.exception(e)
-	    return reverse('basket:summary')
+            return reverse('basket:summary') # whaaaat
```
Comment (monokrome) [inline paypal/express/views.py]: @sjolicoeur Can we get rid of this comment?

File: paypal/express/views.py [hunk 3]
```diff
@@ -212,11 +213,11 @@ def get(self, request, *args, **kwargs):
...
             return HttpResponseRedirect(reverse('basket:summary'))
-
+        else:
+            self.request.basket = kwargs['basket']
         logger.info(
-            "Basket #%s - showing preview with payer ID %s and token %s",
-            kwargs['basket'].id, self.payer_id, self.token)
-
+            "Basket #%s  showing preview with payer ID %s and token %s",
+            kwargs['baset'].id, self.payer_id, self.token)
```
Comment (tpgandhe) [inline paypal/express/views.py]: typo in kwargs['basket']

File: paypal/express/views.py [hunk 4]
```diff
@@ -374,18 +402,32 @@ def get_shipping_method(self, basket, shipping_address=None, **kwargs):
...
-        # Instantiate a new FixedPrice shipping method instance
+        code = self.checkout_session.shipping_method_code(basket) # why is code none for valid orders and showing the not shipping
+        shipping_method = Repository().get_default_shipping_method(
+            user=self.request.user, basket=basket,
+            shipping_addr=shipping_address, request=self.request)
+
+
+        if shipping_address.country.pk not in [country.pk for country in \
+                                        shipping_method.countries.all()]:
+            countries = ", ".join([country.pk for country in \
```
Comment (tpgandhe) [inline paypal/express/views.py]: Can we assign a variable for shipping_method.countires.all() so we don't have to make the query twice :)

**Comment by tpgandhe** [comment]
@sjolicoeur  : Thanks .. :cake:

**Pull request merged**